Tools/environment_install/install-prereqs-openSUSE-Tumbleweed.sh

```diff
@@ -0,0 +1,221 @@
+while getopts "yq" opt; do
+    case "$opt" in
+        \?)
+            exit 1
+            ;;
+        y)  ASSUME_YES=true
+            ;;
+        q)  QUIET=true
+            ;;
+    esac
+done
+
+ZYPPER="sudo zypper in --no-recommends"
+if $ASSUME_YES; then
+    ZYPPER="sudo zypper in -y --no-recommends"
+fi
+PIP3=pip3
+if $QUIET; then
+    PIP3="pip3 -q"
+fi
+
+function package_is_installed() {
+    rpm -q ModemManager &>/dev/null
+}
+
+function heading() {
+    echo "$sep"
+    echo $*
+    echo "$sep"
+}
+
+#Install python3 1st, as openSUSE TW WSL does not come with preinstalled python
+$ZYPPER python3 || echo "Check zypper output for errors"
+
+#As tumbleweed can have many python versions, we need to know which one is currently active
+PYPKGVER=python$(python3 --version | cut -d' ' -f2 | awk -F. '{print $1$2}')
+
+BASE_PKGS="patterns-devel-base-devel_basis ccache git axel valgrind screen gcc-c++ xterm free-ttf-fonts sfml2-devel zip glibc-devel-static rsync"
+SITL_PKGS="${PYPKGVER}-pip ${PYPKGVER}-devel ${PYPKGVER}-setuptools ${PYPKGVER}-wheel ${PYPKGVER}-lxml ${PYPKGVER}-pyaml ${PYPKGVER}-wxPython ${PYPKGVER}-pyparsing ${PYPKGVER}-opencv ${PYPKGVER}-numpy ${PYPKGVER}-scipy ${PYPKGVER}-matplotlib"
+
+PYTHON_PKGS="future lxml pymavlink MAVProxy pexpect argparse pyparsing geocoder pyserial empy==3.3.4 ptyprocess dronecan"
+PYTHON_PKGS+=" flake8 junitparser pygame intelhex psutil pyyaml"
+# GNU Tools for ARM Embedded Processors
+# (see https://launchpad.net/gcc-arm-embedded/)
+ARM_ROOT="gcc-arm-none-eabi-10-2020-q4-major"
+ARM_TARBALL="$ARM_ROOT-x86_64-linux.tar.bz2"
+ARM_TARBALL_URL="https://firmware.ardupilot.org/Tools/STM32-tools/$ARM_TARBALL"
+
+ARM_LINUX_ROOT=gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf
+ARM_LINUX_GCC_URL="https://releases.linaro.org/components/toolchain/binaries/7.5-2019.12/arm-linux-gnueabihf/gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz"
+ARM_LINUX_TARBALL="gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz"
+
+function maybe_prompt_user() {
+    if $ASSUME_YES; then
+        return 0
+    else
+        read -p "$1"
+        if [[ $REPLY =~ ^[Yy]$ ]]; then
+            return 0
+        else
+            return 1
+        fi
+    fi
+}
+
+sudo usermod -a -G dialout "$USER"
+
+$ZYPPER $BASE_PKGS $SITL_PKGS || echo "Check zypper output for errors"
+
+python3 -m venv "$HOME"/venv-ardupilot
+
+SHELL_LOGIN=".profile"
+# activate it:
+SOURCE_LINE="source $HOME/venv-ardupilot/bin/activate"
+$SOURCE_LINE
+
+if ! grep -Fxq "$SOURCE_LINE" ~/.bashrc; then 
+    if [[ -z "${DO_PYTHON_VENV_ENV}" ]] && maybe_prompt_user "Make ArduPilot venv default for python [N/y]?" ; then
+        DO_PYTHON_VENV_ENV=1
+    fi
+
+    if [[ $DO_PYTHON_VENV_ENV -eq 1 ]]; then
+        echo $SOURCE_LINE >> ~/.bashrc
+    fi
+fi
+
+$PIP3 install -U pip setuptools wheel
+$PIP3 install -U attrdict3
+$PIP3 install -U $PYTHON_PKGS
+
+(
+    cd /usr/lib64/ccache
+    if [ ! -f arm-none-eabi-g++ ]; then
```

Comment:
Looks like a for loop is waiting to be born here.

Probably also in the source script....

Reply:
done for this script only